src/battle-text-parser.ts

```diff
@@ -226,20 +230,16 @@ class BattleTextParser {
 
 	pokemonName = (pokemon: string) => {
 		if (!pokemon) return '';
-		if (!pokemon.startsWith('p1') && !pokemon.startsWith('p2')) return `???pokemon:${pokemon}???`;
+		if (!pokemon.startsWith('p1') && !pokemon.startsWith('p2') && !pokemon.startsWith('p3') && !pokemon.startsWith('p4')) return `???pokemon:${pokemon}???`;
 		if (pokemon.charAt(3) === ':') return pokemon.slice(4).trim();
 		else if (pokemon.charAt(2) === ':') return pokemon.slice(3).trim();
 		return `???pokemon:${pokemon}???`;
 	};
 
 	pokemon(pokemon: string) {
 		if (!pokemon) return '';
-		let side;
-		switch (pokemon.slice(0, 2)) {
-		case 'p1': side = 0; break;
-		case 'p2': side = 1; break;
-		default: return `???pokemon:${pokemon}???`;
-		}
+		let side = pokemon.slice(0, 2);
+		if (!['p1', 'p2', 'p3', 'p4'].includes(side)) return `???pokemon:${pokemon}???`;
 		const name = this.pokemonName(pokemon);
 		const template = BattleText.default[side === this.perspective ? 'pokemon' : 'opposingPokemon'];
```

Comment:
Shouldn't allies get different text than foes? I suppose that can be left for a future PR.

Reply:
No, that's an easy enough fix for this one. Something like
```
const teamOf = {'p1': 0, 'p2': 1, 'p3': 0, 'p4': 1};
const isOpposing = teamOf[side] === teamOf[this.perspective];
const template = BattleText.default[isOpposing ? 'pokemon' : 'opposingPokemon'];
```
When supporting other game types with different team setups like Battle Royal we can just use different values for `teamOf`;

Reply:
We already have an `ally` functiom somewhere in the text parser, don't we?